src/overlays/actors/ovl_Obj_Oshihiki/z_obj_oshihiki.c

```diff
@@ -27,63 +67,661 @@ const ActorInit Obj_Oshihiki_InitVars = {
+    this->dyna.actor.gravity = -1.0f;
+    ObjOshihiki_SetGround(this, globalCtx);
+    this->actionFunc = ObjOshihiki_Fall;
+}
+
+void ObjOshihiki_Fall(ObjOshihiki* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    this->stateFlags |= BLOCK_FALL;
+    if (fabsf(this->dyna.unk_150) > 0.001f) {
+        this->dyna.unk_150 = 0.0f;
+        player->stateFlags2 &= ~0x10;
+    }
+    Actor_MoveForward(&this->dyna.actor);
+    if (ObjOshihiki_CheckGround(this, globalCtx)) {
+        if (this->floorBgIds[this->highestFloor] == 0x32 /*BGCHECK_SCENE*/) {
+            ObjOshihiki_SetupOnScene(this, globalCtx);
+        } else {
+            ObjOshihiki_SetupOnActor(this, globalCtx);
+        }
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_BLOCK_BOUND);
+        Audio_PlayActorSound2(&this->dyna.actor, func_80041F34(&globalCtx->colCtx, this->floorPolys[this->highestFloor],
+                                                               this->floorBgIds[this->highestFloor]) +
+                                                     SFX_FLAG);
+    }
+}
+
+void ObjOshihiki_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    ObjOshihiki* this = THIS;
+
+    this->stateFlags &= ~(SETUP_FALL | BLOCK_FALL | SETUP_PUSH | BLOCK_PUSH | SETUP_ON_ACTOR | BLOCK_ON_ACTOR |
+                          SETUP_ON_SCENE | BLOCK_ON_SCENE);
+    this->stateFlags |= MOVE_UNDER;
+
+    if (this->timer > 0) {
+        this->timer--;
+    }
+
+    this->dyna.actor.posRot.rot.y = this->dyna.unk_158;
+
+    this->yawSin = Math_Sins(this->dyna.actor.posRot.rot.y);
+    this->yawCos = Math_Coss(this->dyna.actor.posRot.rot.y);
+
+    if (this->actionFunc != NULL) {
+        this->actionFunc(this, globalCtx);
+    }
+}
+
+void ObjOshihiki_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    ObjOshihiki* this = THIS;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_obj_oshihiki.c", 1289);
+    if (ObjOshihiki_MoveWithBlockUnder(this, globalCtx)) {
+        Matrix_Translate(this->underDistX * 10.0f, 0.0f, this->underDistZ * 10.0f, 1);
+    }
+    this->stateFlags &= ~MOVE_UNDER;
+    func_80093D18(globalCtx->state.gfxCtx);
+    gSPSegment(oGfxCtx->polyOpa.p++, 0x08, SEGMENTED_TO_VIRTUAL(this->texture));
+
+    gSPMatrix(oGfxCtx->polyOpa.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_obj_oshihiki.c", 1308),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+    switch (globalCtx->sceneNum) {
+        case SCENE_YDAN:
+        case SCENE_DDAN:
+        case SCENE_BMORI1:
+        case SCENE_HIDAN:
+        case SCENE_MIZUSIN:
+        case SCENE_JYASINZOU:
+        case SCENE_HAKADAN:
+        case SCENE_MEN:
+            gDPSetEnvColor(oGfxCtx->polyOpa.p++, this->color.r, this->color.g, this->color.b, 255);
+            break;
+        default:
+            gDPSetEnvColor(oGfxCtx->polyOpa.p++, mREG(0xD), mREG(0xE), mREG(0xF), 255);
```

Comment:
Regs are normally numbered as decimal
```suggestion
            gDPSetEnvColor(oGfxCtx->polyOpa.p++, mREG(13), mREG(14), mREG(15), 255);
```